src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1568,6 +1574,18 @@ public override BoundNode VisitCall(BoundCall node)
                 ReplayReadsAndWrites(localFunc, node.Syntax, writes: true);
             }
 
+            if (MethodEnsuresTrueWhenExits(method) && arguments.Length > 0)
+            {
+                VisitTrueWhenExits(condition: arguments[0]);
+            }
+
+            if (MethodReturnsTrueWhenNotNull(method) && arguments.Length > 0)
+            {
+                VisitTrueWhenNotNull(arguments[0], method.ReturnType.TypeSymbol);
+                _result = method.ReturnType;
+                return null;
```

Comment:
Are the last two statements necessary?

Reply:
We need to return, since the assertion below would be violated. 
We need to set the result, otherwise another assertion breaks (we check that the result is not left as invalid type).